src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10476,14 +10477,27 @@ private ExpressionSyntax ParseTermWithoutPostfix(Precedence precedence)
                 case SyntaxKind.StackAllocKeyword:
                     return this.ParseStackAllocExpression();
                 case SyntaxKind.DelegateKeyword:
+                    if (this.IsPossibleLambdaExpression(precedence))
+                    {
+                        return this.ParseLambdaExpression();
+                    }
                     return this.ParseAnonymousMethodExpression();
                 case SyntaxKind.RefKeyword:
+                    if (this.IsPossibleLambdaExpression(precedence))
+                    {
+                        return this.ParseLambdaExpression();
+                    }
```

Comment:
interesting ambiguity.  if we have `ref x () =>`  is that a `ref` to this lambda, or a lambda that returns by-ref.  Given that the former seems fairly meaningless semantically (correct me if i'm wrong), it seems safe to presume this is the latter.

Reply:
can you add comment on this what case this is matching?